pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/constant/ConstantMVForwardIndexReader.java

```diff
@@ -45,6 +45,9 @@ public DataType getStoredType() {
 
   @Override
   public int getDictIdMV(int docId, int[] dictIdBuffer, ForwardIndexReaderContext context) {
+    if (dictIdBuffer.length == 0) {
```

Comment:
This is not needed when `maxNumberOfMultiValues` is properly set

Reply:
This looks like a code smell and is better to be more defensive. There is no guarantee that a caller to this method knows how `maxNumberOfMultiValues` influences the behavior of this method.

Reply:
Discussed with @Jackie-Jiang  offline. Looks like `dictIdBuffer.length > 0` is the expected invariant for this method due to how MV column values are stored in the bitmap. I have added some javadocs so that future contributors are aware of this. Please do take a look. Thanks!